### PR TITLE
[21.09] BugFix: Update social-auth-core library version to fix OIDC problem

### PR DESCRIPTION
**Pull request by IslamHeggy**
This issue contains a full description of the error and the resolution. #12955 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -69,7 +69,7 @@ PyYAML = "*"
 refgenconf = ">=0.12.0"
 requests = "*"
 Routes = "*"
-social-auth-core = {version = "==3.3.0", extras = ["openidconnect"]}
+social-auth-core = {version = "==4.0.3", extras = ["openidconnect"]}
```
Comment (mvdbeek) [inline pyproject.toml:72]: @nsoranzo Do we also need to add this to `lib/galaxy/dependencies/pinned-requirements.txt` and `lib/galaxy/dependencies/dev-requirements.txt` ?

Comment (nsoranzo) [reply pyproject.toml:72]: Yep, I was going to run Poetry locally and push the relevant changes. Or we can merge and wait for https://github.com/galaxyproject/galaxy/pull/12916 to pick it up.

Comment (mvdbeek) [reply pyproject.toml:72]: Does https://github.com/galaxyproject/galaxy/pull/12916 run against 21.09 as well ?

Comment (nsoranzo) [reply pyproject.toml:72]: Ops, no, I didn't notice the target branch, sorry.

Comment (IslamHeggy) [reply pyproject.toml:72]: @mvdbeek @nsoranzo Please let me know if I need to update the mentioned files above

Comment (nsoranzo) [reply pyproject.toml:72]: After checking that the change is compatible with the other requirements in these files, I've pushed the change to this PR.

**Review by nsoranzo — APPROVED** [review]

**Comment by mvdbeek** [comment]
Thanks @IslamHeggy!

**Pull request merged**